Src/Test.UnitTests.Sarif.PatternMatcher/SpamTestRule.cs

```diff
@@ -29,11 +29,11 @@ internal class SpamTestRule : Skimmer<AnalyzeContext>, IOptionsProvider
         internal static Random s_random = new Random(s_seed);
 
         [ThreadStatic]
-        internal static TestRuleBehaviors s_testRuleBehaviors;
+        private static TestRuleBehaviors s_testRuleBehaviors;
```

Comment:
>private [](http://example.com/codeflow?start=8&length=7)

You are going to break upstream consumers of this property, I believe. We shouldn't do it but this data is visible to some SPAM things, I think, check the internals visible to attribute data.

The 'right' thing to do would be to get over it and just elevate this to a public static property.

Reply:
Ok, now I remember, the VS auto fix was to create a property and s_testRuleBehaviors turned into the backing field which is why it could become private but the property became S_testRuleBehaviors.

But, I have restored everything back and for now there is a warning about it:
warning CS0649: Field 'SpamTestRule.s_testRuleBehaviors' is never assigned to, and will always have its default value